base/div.jl

```diff
@@ -83,6 +91,10 @@ rem(x, y, ::RoundingMode{:Up}) = mod(x, -y)
 rem(x, y, r::RoundingMode{:Nearest}) = x - y*div(x, y, r)
 rem(x::Integer, y::Integer, r::RoundingMode{:Nearest}) = divrem(x, y, r)[2]
 
+function rem(x, y, ::typeof(RoundFromZero))
+    rem(x, y, ifelse(signbit(x) == signbit(y), RoundUp, RoundDown))
```

Comment:
I think this is a type instability, might be better to branch outside?
```
signbit(x) == signbit(y) ? rem(x, y, RoundUp) : rem(x, y, RoundDown)
```
Although when I tried to benchmark that, I didn't see a timing difference between the two.

Reply:
I tested with `BigInt`s and I also see no difference in performance (compiled Julia with `JULIA_CPU_TARGET=native` and `MARCH=native`).

I tend to think you are right, though. The compiler seems to be able to handle that type instability, but there might be some corner cases in which it doesn't.